src/librustc_mir/dataflow/mod.rs

```diff
@@ -185,12 +185,11 @@ impl<'a, 'tcx: 'a, BD> DataflowAnalysis<'a, 'tcx, BD> where BD: BitDenotation
         let mut temp = IdxSetBuf::new_empty(self.flow_state.sets.bits_per_block);
         let mut propcx = PropagationContext {
             builder: self,
-            changed: true,
         };
-        while propcx.changed {
-            propcx.changed = false;
-            propcx.walk_cfg(&mut temp);
-        }
+        let mut dirty_queue: WorkQueue<mir::BasicBlock> =
```

Comment:
Nit: we should probably move the `dirty_queue` creation into `walk_cfg`; or else merge `walk_cfg` and this method. But anyway, doesn't matter.